source/docs/software/troubleshooting/robots-shouldnt-quit.rst

```diff
@@ -0,0 +1,415 @@
+Robots Should Not Quit, But Yours Did!
+======================================
+
+``Robots should not quit, but yours did!``
+
+When your robot code hits an unexpected error, you will see this message show up in some console output (Driver Station or rioLog). You'll probably also notice your robot abruptly stop, or possibly never move. These unexpected errors are called *unhandled exceptions*.
+
+When an unhandled exception occurs, it means that your code has one or more bugs which need to be fixed.
+
+This article will explore some of the tools and techniques involved in finding and fixing those bugs.
+
+What's a Stack Trace?
+---------------------
+
+The ``Robots should not quit`` message is a signal that a *stack trace* has been printed out. 
+
+In C++ and Java, a `stack <https://en.wikipedia.org/wiki/Call_stack>`_ data structure is used to store information about which function or method is currently being executed.
+
+A *stack trace* prints information about what was on this stack when the unhandled exception occurred.This points you to the lines of code which were running just before the problem happened. While it doesn't always point you to the *exact root cause* of your issue, it's usually the best place to start looking.

```

Comment:
Is the emphasis on `exact root cause` necessary/consistent with the rest of the doc writing style?

Reply:
I'll got take a peek at the style guide, I'm not entirely sure. I'm definitely not tied to having it there, it's more a reflection of where I'd place my verbal emphasis if I were saying it out loud. 

My end goal would really just be to drive home the main point of the whole article: a stack trace is a _tool to solve your problem_, not a solution. I've found new folks often get confused (and discouraged) when they see an experienced person look at a stack trace (or similar), come up with an answer, and assume that the answer must have come from within said stack trace.

TL;DR probably doesn't have to be there.

Reply:
Okay, fair enough :)